service/src/test/java/uk/gov/hmcts/reform/fpl/service/DraftCMOServiceTest.java

```diff
@@ -0,0 +1,118 @@
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicList;
+
+import java.time.LocalDate;
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.fpl.utils.CaseDataGeneratorHelper.createHearingBooking;
+
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = {JacksonAutoConfiguration.class, DateFormatterService.class, DraftCMOService.class})
+public class DraftCMOServiceTest {
+
+    @Autowired
+    private DraftCMOService draftCMOService;
+
+    private final LocalDate date = LocalDate.now();
+
+    @Test
+    void getHearingDatesDynamic_shouldReturnHearingDateDynamicList() {
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(ImmutableMap.of(
+                "hearingDetails", createHearingBookings(date))
+            )
+            .build();
+
+        DynamicList hearingList = draftCMOService.getHearingDatesDynamic(caseDetails);
+
+        assertThat(hearingList.getListItems().get(0).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date.plusDays(5)));
+
+        assertThat(hearingList.getListItems().get(1).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date.plusDays(2)));
+
+        assertThat(hearingList.getListItems().get(2).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date));
+
+    }
+
+    @Test
+    void getHearingDatesDynamicShouldFillTheHearingDatesListWithPreviousSelectedValue() {
+
+        CaseDetails caseDetails = CaseDetails.builder()
+            .data(ImmutableMap.of(
+                "hearingDetails", createHearingBookings(date),
+                "caseManagementOrder", createCaseManagementOrder(draftCMOService.convertDate(date.plusDays(2)))
+                )).build();
+
+        DynamicList hearingList = draftCMOService.getHearingDatesDynamic(caseDetails);
+
+        assertThat(hearingList.getListItems().get(0).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date.plusDays(5)));
+
+        assertThat(hearingList.getListItems().get(1).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date.plusDays(2)));
+
+        assertThat(hearingList.getListItems().get(2).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date));
+    }
+
+    @Test
+    void shouldReturnHearingDateDynamicListIfHearingDatesNotNull() {
+        DynamicList hearingList = draftCMOService.buildDynamicListFromHearingDetails(
+            createHearingBookings(date));
+
+        assertThat(hearingList.getListItems().get(0).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date.plusDays(5)));
+
+        assertThat(hearingList.getListItems().get(1).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date.plusDays(2)));
+
+        assertThat(hearingList.getListItems().get(2).getLabel())
+            .isEqualTo(draftCMOService.convertDate(date));
+    }
+
+    @Test
+    void convertDateShouldReturnDateInUKFormat() {
```

Comment:
We are re-testing dateFormatterService here. Not needed. You can test your date in the asserts above.